python/ray/data/_internal/execution/operators/task_pool_map_operator.py

```diff
@@ -82,7 +91,7 @@ def _add_bundled_input(self, bundle: RefBundle):
                 2 * data_context._max_num_blocks_in_streaming_gen_buffer
             )
 
-        gen = map_task.options(**ray_remote_args).remote(
+        gen = self._map_task.options(**ray_remote_args).remote(
```

Comment:
let's add  `assert "num_gpus" not in ray_remote_args` to prevent this issue from happening again? 
also, maybe rename the variable to `dynamic_remote_args`.

Reply:
Can't add an assertion, b/c this set includes static args as well we call base function to produce the union of static and dynamic